lib/src/facts/posix/uptime_resolver.cc

```diff
@@ -43,6 +48,22 @@ namespace facter { namespace facts { namespace posix {
 
     int64_t uptime_resolver::get_uptime()
     {
+        // We first try to get the system boot time by reading it directly from
+        // the utmp file using the getutxid system call.
+        LOG_DEBUG("Attempting to calculate the uptime from the utmp file...");
+        struct utmpx query;
+        query.ut_type = BOOT_TIME;
+        setutxent();
+        struct utmpx* ent = getutxid(&query);
```

Comment:
Maybe we could completely take out the `uptime` command in our POSIX resolver and just delegate to getutxid instead. I've tested this code on OSX 10.13, RHEL 7, Ubuntu Bionic, Debian 9 and all of the AIX and 64-bit Solaris platforms I mentioned in the PR comments (will update soon) and have found that it seems to return the right information for all of them.

NOTE: Testing was done by copy-pasting this to a separate CPP file then copying it over to the VMs and compiling it on there.